### ISSUE TITLE
azure-core-tracing-opentelemetry should not be taking dependencies on SNAPSHOT versions

### ISSUE THREAD

**Issue by JimSuplizio** [Azure/azure-sdk-for-java#6301]
We should not be taking dependencies on SNAPSHOT versions, there are two in here io.opentelemetry:opentelemetry-api and io.opentelemetry:opentelemetry-sdk. We could not release with these dependencies. Further, I believe that unless azure-core-tracing-opentelemetry's version was also -SNAPSHOT, that Maven would reject the package.

We should not be adding tags to the individual component files. This overrides the repository settings defined further up in the hierarchy and bypasses our cache.

There's also one other big issue with the SNAPSHOT version dependencies is that Maven is going to try and query to pull these every time which will cause transient errors. I'm actually running into a transient error on my local machine because it this which built earlier and is failing now.

The PR That introduced this is https://github.com/Azure/azure-sdk-for-java/pull/6131

**Comment by samvaity** [comment]
Blocked by https://github.com/open-telemetry/opentelemetry-java/issues/674

**Comment by joshfree** [comment]
Tagging this as `tracking-external-issue`.

@jimsuplizio All 4 language repos will be shipping azure-core-tracing-opentelemetry preview plugins that depend on v0.2-alpha opentelemetry SDKs in the December Update.  The engineering system will need to be able to accommodate this until OpenTelemetry has proper releases in early 2020.

/cc @lmolkova as fyi

**Comment by joshfree** [comment]
> unless azure-core-tracing-opentelemetry's version was also -SNAPSHOT, that Maven would reject the package.

@samvaity can you please experiment with this outside of the devops pipeline and confirm/deny maven would reject the package versioning?  This should be fixed on our side before we get to the `[2019] December` code complete.

**Comment by JimSuplizio** [comment]
>> The engineering system will need to be able to accommodate this until OpenTelemetry has 
>> proper releases in early 2020.
//CC @weshaggard @mitchdenny 
@joshfree There's nothing we can do to actually accommodate this on the EngSys side. The very nature of having a SNAPSHOT dependency is going to bypass any cache we have because for SNAPSHOTs it needs to check for a newer versions **every time** this builds. Further, as I'd previously called out, this can cause transient failures. I was actually seeing intermittent failures trying to pull these dependencies when building locally at the time I'd created this issue. If we're waiting for an ALPHA release (or really any actual usable release) then I'd recommend removing this module from pom.client.xml until there is one.

**Comment by samvaity** [comment]
@JimSuplizio and @joshfree:
 @carlosalberto just confirmed that we can have the Maven packages for Otel max by tomorrow. I can have a follow-up PR to look up the released version in our pom as soon as they release.

>@samvaity I will work on this in a bit. So I'd guess we should have packages rather soon (max. tomorrow, I'd expect). I will try to to that through Bintray, btw
Well, actually tomorrow is probably a good estimate, as I know we need to request inclusion into Maven, which can take between a few hours and 1 day.
